src/server/controllers/plugins/Push.ts

```diff
@@ -0,0 +1,96 @@
+const request = require('request');		
+
+import { OutputPlugin } from '../Output';
+
+type pluginDependencies = {
+    apiKey: string,
+    appId: string,
+    content: string
+}
+
+export class Push extends OutputPlugin {
+
+    private dependencies: pluginDependencies = {
+        apiKey: null,
+        appId: null,
+        content: null
+    };
+
+    public async propagate(): Promise<string | string[]> {
+        const url = 'https://onesignal.com/api/v1/notifications';
+        const { apiKey, appId, content } = this.dependencies;
+
+        const promises = [];
+        const outputs = [];
+        const sendMessage = (appId, content, apiKey) => new Promise((resolve, reject) => {
+                request.post({
+                	url: url,
+                	body: JSON.stringify({
+	        			app_id: appId,
+	        			contents: {
+	        				en: content
+	        			},
+	        			included_segments: ["All"]
+                	}),
+                	headers: {
+                		'content-type': 'application/json',
+                		'Authorization': "Basic " + apiKey
+                	}
+                }, function (err, response) {
+                    if (err) {
+                        reject(err);
+                    } else {
+                        resolve(response);
+                    }
+                });
+            }
+        );
+        if(this.numIter === null) this.numIter = 1;
+        for (let i = 0; i < this.numIter; i++) {
```

Comment:
Is this necessary when all inputs can only be single values?

Reply:
I decided to leave that in b/c in the future we could add some iterable fields (different messages for different segments etc) but are probably out of the scope of the event. LMK if you still want me to remove